frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx

```diff
@@ -245,20 +258,27 @@ const CustomNodeTable: React.FC<CustomNodeTableProps> = ({
     const promises = [];
 
     promises.push(...makeLabelNodesRequest(selectedData));
+
     // intentionally keeping the taint logic as its required in 4.3 and will be handled with checkbox selection
     // promises.push(...makeTaintNodesRequest(selectedData));
 
     const ocsObj = _.cloneDeep(ocsRequestData);
     ocsObj.spec.storageDeviceSets[0].dataPVCTemplate.spec.storageClassName = storageClass;
-    promises.push(k8sCreate(OCSServiceModel, ocsObj));
 
+    // Nested Promises failing lint test as well
     Promise.all(promises)
       .then(() => {
-        history.push(
-          `/k8s/ns/${ocsProps.namespace}/clusterserviceversions/${
-            ocsProps.clusterServiceVersion.metadata.name
-          }/${referenceForModel(OCSServiceModel)}/${ocsObj.metadata.name}`,
-        );
+        k8sCreate(OCSServiceModel, ocsObj).then(() => {
+          history.push(
+            `/k8s/ns/${ocsProps.namespace}/clusterserviceversions/${
+              ocsProps.clusterServiceVersion.metadata.name
+            }/${referenceForModel(OCSServiceModel)}/${ocsObj.metadata.name}`,
+          );
+        })
+        .catch((err) => {
+          setProgress(false);
+          setError(err.message);
+        });
```

Comment:
Not required to be changed. if we are handling above via if condition. 

If promise obj is empty we need to block user to select new nodes.

Reply:
@gnehapk can you provide proper details on this one.

Reply:
@cloudbehl We need to handle the scenario where if labeling node api call failed, it should not call k8sCreate. So, its required I feel. Thoughts?